File: cpp/program28.cpp
// program60.c

#include <iostream>
using namespace std;

class Digit
{
    public:
        int iNo;

    Digit(int X)
    {
        iNo = X;
    }

    int CheckDigitFrequency()       
    {
        int iDigit = 0;
        int iSearch = 0;
        int iCount = 0;
        int iTemp = 0;

        cout<<"Enter the Digit you want to search (0 to 9)"<<"\n";
        cin>>iSearch;

        if((iSearch < 1) || (iSearch > 9))
        {
            cout<<"Invalid Input"<<"\n";
            cout<<"NOTE : please enter digit between range (0 to 9 )"<<"\n";
            return 0;
        }

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        iTemp = iNo;

        while(iTemp != 0)
        {
            iDigit = iTemp % 10;
            if(iDigit == iSearch)
            {
                iCount++;
            }
            iTemp = iTemp / 10;
        }
        return iCount;
    }
     
};

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout<<"Enter number"<<"\n";
    cin>>iValue;

    Digit dobj(iValue);

    iRet = dobj.CheckDigitFrequency();
    cout<<"Frequency of the Digit is : "<<iRet<<"\n";

    iRet = dobj.CheckDigitFrequency();
    cout<<"Frequency of the Digit is : "<<iRet<<"\n";

    return 0;
}

File: java/program28.java
import java.util.*;

class Digit 
{
    public int iNo;

    public Digit(int no)
    {
        iNo = no;
    }

    public int CheckFrequency(int iSearch)
    {
        int iDigit = 0,iCount = 0;
        int temp = 0;

        if(iNo < 0)
        {
            iNo = -iNo;
        }

        if((iSearch < 1) || (iSearch > 9))
        {
            System.out.print("Invalid Input \nNote : please enter digit between (0 to 9) \n");
            return 0;
        }

        temp = iNo;
        while(temp != 0)
        {
            iDigit = temp % 10;
            if(iDigit == iSearch)
            {
                iCount++;
            }
            temp = temp / 10;
        }
        return iCount;
    }
}

class program28
{
    public static void main(String arg[])
    {
        int iValue1 = 0,iValue2 = 0,iRet = 0;
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number");
        iValue1 = sobj.nextInt();

        System.out.println("Enter the digit to check frequency between (1 to 9)");
        iValue2 = sobj.nextInt();

        Digit obj = new Digit(iValue1);
        iRet = obj.CheckFrequency(iValue2);

        System.out.println("Frequency of digit is : "+iRet);
    }
}